network/struct.go

```diff
@@ -107,6 +109,7 @@ func NewServerIdentity(public abstract.Point, address Address) *ServerIdentity {
 		Public:  public,
 		Address: address,
 		ID:      ServerIdentityID(uuid.NewV5(uuid.NamespaceURL, url)),
+		Cert:    []byte{},
```

Comment:
```
sid := NewTLSServerIdentity(public,address,cert)
```
vs 
```
sid := NewServerIdentity(public, address)
sid.Cert = cert
```

@Liamsi will maybe have to be the deciding party but to me it's clear which version is shorter / cleaner and have less chance of making weird bug if we forget this `sid.Cert = cert` line somewhere...
After the refactoring PRs, it's my turn to nit pick the code of others now .. ! :D